src/phoenix/core/traces.py

```diff
@@ -290,3 +328,21 @@ def _add_value_to_span_ancestors(
             cumulative_value = parent_span[attribute_name] or 0
             parent_span[attribute_name] = cumulative_value + value
             span_id = parent_span_id
+
+
+def _validate_new_span(
+    new_span: pb.Span,
+    existing_span: Optional[pb.Span] = None,
+) -> None:
+    if not new_span.HasField("start_time"):
+        raise ValueError("start_time is missing")
+    if existing_span is None:
+        return
+    if (
+        existing_span.HasField("start_time")
+        and new_span.HasField("start_time")
+        and existing_span.start_time != new_span.start_time
+    ):
+        raise ValueError(
+            f"invalid start_time, expected={existing_span.start_time}, found={new_span.start_time}"
+        )
```

Comment:
We don't have to add this, but the intent is to allow the rest of the code to make assumptions that would otherwise have no guarantee. It makes no practical difference, but provides a set of "axioms" for inferring correctness on the rest of the code.

Reply:
got it